### PR TITLE
Feature/ubi 234 microservice http comms

### PR DESCRIPTION
**Pull request by shortChangeDev**
#### What changes does this PR contain?
Adds an interface to manage HTTP requests (to microservices) and subscribe to responses

#### Which JIRA ticket ID's are associated with this PR?
UBI-234

#### How was it tested (Steps to reproduce by reviewers)?
Start Docker
Write http requests (PUT, GET, DELETE, POST) from somewhere in the code (calling methods from HTTPClient)
Such as 
```
Debug.Log(Get("products"));
PostJSON("products", "{\"name\":\"New Object added by Murphy\", \"price\":69.69, \"sku\":\"abc-abc-abcd\"}");
Debug.Log(Get("products"));
```
Start the client

#### Notes for reviewers
- 
@ubivius/gamedev

### CODE REVIEW

File: Assets/Scripts/HTTPToolkit/HTTPClient.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        }
+
+        public HttpResponseMessage Delete(string requestUrl)
+        {
+            HttpResponseMessage result = m_client.DeleteAsync(m_endPoint + "/" + requestUrl).Result;
+
+            return result;
+        }
+    }
+}
```
Comment (ysma500) [inline Assets/Scripts/HTTPToolkit/HTTPClient.cs:72]: No new line EOF

File: Assets/Scripts/HTTPToolkit/HTTPClient.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            
+            return response;
+        }
+
+        public HttpResponseMessage Post(string requestUrl, object objToSerialize)
+        {
+            return PostJSON(requestUrl, JsonUtility.ToJson(objToSerialize));
+        }
+
+        public HttpResponseMessage Put(string requestUrl, string jsonString)
```
Comment (ysma500) [inline Assets/Scripts/HTTPToolkit/HTTPClient.cs:38]: Why not PutJSON like the PostJSON function name?

**Review by ysma500 — CHANGES_REQUESTED** [review]
Could there be a test file with a mocked http server created for unit testing with this lib?

**Review by JerPaq — DISMISSED** [review]

File: Assets/Scripts/HTTPToolkit/HTTPClient.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    /// <summary>
+    /// HTTP client custom wrapper
+    /// See https://zetcode.com/csharp/httpclient/ 
+    /// and https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=net-5.0
+    /// </summary>
+    public class HTTPClient : MonoBehaviour
+    {
+        static private readonly HttpClient m_client = new HttpClient();
+
+        [SerializeField] string m_endPoint = "http://localhost:9090";
```
[the remark this answers is not in the document]
Comment (shortChangeDev) [reply Assets/Scripts/HTTPToolkit/HTTPClient.cs:17]: private par défaut, mais je vais le rajouter quand même, merci!

**Review by Daftdoritos — APPROVED** [review]

**Review by ArcticWalrus — APPROVED** [review]
Ça semble bon. On va devoir y apporter des changements quand on ajoute l'authentification aux microservices mais d'ici là c'est bon

**Comment by ArcticWalrus** [comment]
Il reste juste à fix le CI avant

**Review by ysma500 — APPROVED** [review]
LGTM

**Pull request merged**